pkg/operator/ceph/object/user/controller.go

```diff
@@ -334,6 +357,28 @@ func generateUserConfig(user *cephv1.CephObjectStoreUser) admin.User {
 		Keys:        make([]admin.UserKeySpec, 1),
 	}
 
+	if user.Spec.Quotas != nil && user.Spec.Quotas.MaxBuckets != 0 {
+		userConfig.MaxBuckets = &user.Spec.Quotas.MaxBuckets
+	}
```

Comment:
`admin.User` has a `UserQuota` spec inside of it which is the same thing set in the above code. Why are we setting bucket quotas at a different point in the code from the rest of the quotas? Is it not better to just set everything all at once upon user creation? (i.e., move the other quota code here)

Reply:
That's how RGW adminOps apis works, `maxBucket` is not part of quota Spec. And we can set quota(I mean object and size) only after successful user creation, not during the `CreateUser()`. Hence I moved to `setquota()` after the success of user creation.  So I cannot move the quota code here. The setting `MaxBuckets` can be moved to another quota code part, but it requires another additional ModifyUser() api call. So I kept the MaxBuckets during the creation of User and other quotas after user creation.